rlclientlib/logger/http_transport_client.h

```diff
@@ -66,7 +66,13 @@ class http_transport_client : public i_sender
     int join();
 
   private:
-    pplx::task<web::http::status_code> send_request(size_t try_count);
+    // repeat request until success or _max_retries attempted
+    // returns the http::status_code of the final attempt.
+    pplx::task<web::http::status_code> send_request();
+
+    // repeat request until success or _max_retries attempted
+    // returns the http_reponse of the final attempt.
+    pplx::task<http_response> send_request_with_retries(size_t try_count);
```

Comment:
not very familiar with this code but line 103 of this file seems to have a queue of http_request_task, does that mean that we have separate count of retries for each task?

Reply:
http_transport_client manages up to N concurrent upload tasks.  Each individual concurrent upload is a launched pplx::task and is tracked in the collection of task objects you saw.

The current retry count is stored inside http_transport_client::http_request_task, which is the individual object created to track an individual concurrent upload.